### PR TITLE
Removed "core classes first" logic.

### PR DESCRIPTION
**Pull request by paulp**
I see no evidence this is necessary, and it should be wildly
uncontroversial that if we can lose this, we should. Hardcoded filenames
which don't even exist in the repository ("StandardEmbeddings.scala",
"EmbeddedControls.scala") have no place in compiler source. If some
portion of this is believed necessary then better evidence of that
should be offered.

Review by @lrytz.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/Global.scala [hunk 1]
```diff
@@ -1659,46 +1658,6 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
...
-     *
-     */
-    private def coreClassesFirst(files: List[SourceFile]) = {
-      val goLast = 4
-      def rank(f: SourceFile) = {
-        if (f.file.container.name != "scala") goLast
-        else f.file.name match {
-          case "LowPriorityImplicits.scala"   => 2
-          case "StandardEmbeddings.scala"     => 2
-          case "EmbeddedControls.scala"       => 2
```
Comment (VladUreche) [inline src/compiler/scala/tools/nsc/Global.scala:1695]: I remember EmbeddedControls.scala is used in scala-virtualized, better also ping @adriaanm of this change!

Comment (adriaanm) [inline src/compiler/scala/tools/nsc/Global.scala:1695]: thanks for the ping, @VladUreche, I'll make sure it doesn't cause trouble when cutting M6-virtualized, but i'm sure it doesn't need more special treatment than the other special files

Comment (paulp) [inline src/compiler/scala/tools/nsc/Global.scala:1695]: If it does need special treatment, let's expose this in some way that lets scala-virtualized treat it specially, not hardcode filenames from another project into our source, which is wrong on many levels.

**Comment by lrytz** [comment]
I agree the code is ugly. However how do can you be so sure it's no longer needed, the `LowPriorityImplicits` part?

**Comment by paulp** [comment]
I can't be sure, but the burden of proof should be on the person who says it's needed.  Code this ugly is presumptively unnecessary; no test case which shows what it does, nothing at all?

I can tell you what I did: 1) bootstrap without it and 2) compile Predef and LowPriorityImplicits each by themselves, and together in both orders.  I believe that easily fulfills my obligation.

**Comment by lrytz** [comment]
it's possible that the problem left us together with `ScalaObject`

we can put it in and see if we run into initialization problems. the only thing i can say is that the problem was real at the time i fixed it. and that writing a test for it with partest is not possible.

**Comment by paulp** [comment]
I don't mean to dispute that it solved a real issue at the time it was written.  If you can't write a partest test, there are many other things which could be done; even a comment which says "before this commit, attempting to build with the following command line fails" would be very helpful.

**Comment by paulp** [comment]
I'm understanding lrytz's last comment as "LGTM".

**Pull request merged**